Sources/Sentry/SentrySystemEventsBreadcrumbs.m

```diff
@@ -0,0 +1,174 @@
+#import "SentrySystemEventsBreadcrumbs.h"
+#import "SentrySDK.h"
+#import "SentryLog.h"
+
+// all those notifications are not available for tvOS
+#if TARGET_OS_IOS
+#    import <UIKit/UIKit.h>
+#endif
+
+@implementation SentrySystemEventsBreadcrumbs
+
+- (void)start
+{
+#if TARGET_OS_IOS
+    UIDevice *currentDevice = [UIDevice currentDevice];
+    if (currentDevice == nil) {
+        [SentryLog logWithMessage:@"currentDevice is null, it won't be able to record battery breadcrumbs."
+                         andLevel:kSentryLogLevelDebug];
+        return;
+    }
+    
+    [self initBatteryObserver:currentDevice];
+    [self initOrientationObserver:currentDevice];
+    [self initKeyboardVisibilityObserver];
+    [self initScreenshotObserver];
+    
+#else
+    [SentryLog logWithMessage:@"NO UIKit -> [SentrySystemEventsBreadcrumbs.start] does nothing."
+                     andLevel:kSentryLogLevelDebug];
+#endif
+}
+
+#if TARGET_OS_IOS
+- (void)initBatteryObserver:(UIDevice*)currentDevice
+{
+    if (currentDevice.batteryMonitoringEnabled == NO) {
+        currentDevice.batteryMonitoringEnabled = YES;
+    }
+    
+    NSNotificationCenter *defaultCenter = [NSNotificationCenter defaultCenter];
+    
+    // https://developer.apple.com/documentation/uikit/uidevicebatteryleveldidchangenotification
```

Comment:
`l`: I don't think we need this comment to the Apple docs here.

Reply:
the definition of the constants don't have documentation (javadocs?!), so one would like to know what this is about, the link helps a lot if you have no clue

Reply:
FYI Option + Left Click opens quick help. So no need for the link IMO.

Reply:
@philipphofmann sorry I don't follow, if the fields don't have its own docs in-code, how the quick help would be useful?

Reply:
If you jump to the definition and you don't see any comments that doesn't state that there is no documentation for it. If you open quick help it shows the same documentation as the one when clicking on the link.

Reply:
@philipphofmann can you share a screenshot? I really don't get your point or I don't see any useful things other than the field declaration.
![Screenshot 2020-06-09 at 09 30 44](https://user-images.githubusercontent.com/5731772/84118926-f6ddb380-aa33-11ea-8642-7a019984ecdf.png)

Reply:
I replaced the urls with the comments from the website